api/v1alpha4/metal3machinetemplate_types.go

```diff
@@ -23,6 +23,12 @@ import (
 // Metal3MachineTemplateSpec defines the desired state of Metal3MachineTemplate
 type Metal3MachineTemplateSpec struct {
 	Template Metal3MachineTemplateResource `json:"template"`
+
+	// When set to True, CAPM3 Machine controller will
+	// pick the same pool of BMHs' that were released during the upgrade operation.
+	// +kubebuilder:default=false
```

Comment:
How about defaulting `NodeReuse` using webhook instead of kubebuilder defaulting?

Reply:
Would you mind elaborating more, what is the objection to not using kubebuilder but the webhook option?

Reply:
This is more of a personal taste. Since we have defaulting function in every webhook by default. For example one for the metal3machinetemplate can be found here: https://github.com/metal3-io/cluster-api-provider-metal3/blob/841ab05ddfa413f7898e2c0078c67916dd83d7e8/api/v1alpha4/metal3machinetemplate_webhook.go#L36

IMO it would make more sense to do all defaulting and also field validation per object in one place. But this is something that you may choose, I am fine with kubebuilder defaulting.

Reply:
I would prefer to keep it as it is if there are no strong objections.